cartridges/custom_cartridge/cartridge/controllers/Account.js

```diff
@@ -0,0 +1,368 @@
+
+        var registrationForm = server.forms.getForm('profile');
+
+        // form validation
+        if (registrationForm.customer.email.value.toLowerCase()
+            !== registrationForm.customer.emailconfirm.value.toLowerCase()
+        ) {
+            registrationForm.customer.email.valid = false;
+            registrationForm.customer.emailconfirm.valid = false;
+            registrationForm.customer.emailconfirm.error =
+                Resource.msg('error.message.mismatch.email', 'forms', null);
+            registrationForm.valid = false;
+        }
+
+        if (registrationForm.login.password.value
+            !== registrationForm.login.passwordconfirm.value
+        ) {
+            registrationForm.login.password.valid = false;
+            registrationForm.login.passwordconfirm.valid = false;
+            registrationForm.login.passwordconfirm.error =
+                Resource.msg('error.message.mismatch.password', 'forms', null);
+            registrationForm.valid = false;
+        }
+
+        if (!CustomerMgr.isAcceptablePassword(registrationForm.login.password.value)) {
+            registrationForm.login.password.valid = false;
+            registrationForm.login.passwordconfirm.valid = false;
+            registrationForm.login.passwordconfirm.error =
+                Resource.msg('error.message.password.constraints.not.matched', 'forms', null);
+            registrationForm.valid = false;
+        }
+
+        // setting variables for the BeforeComplete function
+        var registrationFormObj = {
+            firstName: registrationForm.customer.firstname.value,
+            lastName: registrationForm.customer.lastname.value,
+            phone: registrationForm.customer.phone.value,
+            email: registrationForm.customer.email.value,
+            emailConfirm: registrationForm.customer.emailconfirm.value,
+            password: registrationForm.login.password.value,
+            passwordConfirm: registrationForm.login.passwordconfirm.value,
+            validForm: registrationForm.valid,
+            form: registrationForm
+        };
+
+        if (registrationForm.valid) {
+            res.setViewData(registrationFormObj);
+
+            var HookManager = require('dw/system/HookMgr');
+
+            //send post request with customer`s datata to create new customer
+            var UUIDUtils = require('dw/util/UUIDUtils');
+            var integrationId = UUIDUtils.createUUID();
+
+            var result;
+            if (HookManager.hasHook('app.register.requestCustomerToExternalService')) {
+                result = HookManager.callHook(
+                    'app.register.requestCustomerToExternalService',
+                    'requestCustomerToExternalService',
+                    "POST",
+                    registrationFormObj,
+                    integrationId
+                );
+            }
+            if (!result.ok) {
+                return;
```

Comment:
You should have some response returned to the front-end with a generic error like "Something went wrong. Please try again later."

Reply:
done